lib/Sema/ConstraintSystem.cpp

```diff
@@ -2626,6 +2669,17 @@ ConstraintSystem::getTypeOfMemberReference(
                                                        functionType, locator);
     // FIXME: Verify ExtInfo state is correct, not working by accident.
     FunctionType::ExtInfo info;
+
+    if (Context.LangOpts.hasFeature(Feature::InferSendableFromCaptures)) {
+      if (isPartialApplication(*this, funcDecl, locator) &&
+          isSendableType(DC->getParentModule(), baseOpenedTy)) {
+        // Add @Sendable to functions without conditional conformances
+        functionType = functionType->withExtInfo(functionType->getExtInfo().withConcurrent())->getAs<FunctionType>();
```

Comment:
No need to use `getAs<FunctionType>()` here either.

Reply:
functionType->withExtInfo() produces AnyFunctionType but it looks like functionType is expected to be FunctionType? 

```
 // Strip off the 'self' parameter
    auto *functionType = fullFunctionType->getResult()->getAs<FunctionType>();
```

Reply:
Simplify cast https://github.com/apple/swift/pull/67498/files#diff-d57f6f2d466f1f207cdf2dbd0396a1dbb2caba7b6828ebdb9954948f0ac2b499R2667 instead of `FunctionType` to `AnyFunctionType`